agent/connect/ca/provider_vault_test.go

```diff
@@ -464,94 +530,168 @@ func TestVaultCAProvider_SignLeaf(t *testing.T) {
-			}
+		attr1 := &VaultTokenAttributes{
+			RootPath:         "pki-root",
+			IntermediatePath: "pki-intermediate",
+			ConsulManaged:    true,
+			WithSudo:         withSudo,
+		}
+		token1 := CreateVaultTokenWithAttrs(t, testVault1.client, attr1)
 
-			provider2, testVault2 := testVaultProviderWithConfig(t, true, map[string]interface{}{
-				"LeafCertTTL":    "1h",
-				"PrivateKeyType": tc.CSRKeyType,
-				"PrivateKeyBits": tc.CSRKeyBits,
-			})
-			defer testVault2.Stop()
+		provider1 := createVaultProvider(t, true, testVault1.Addr, token1, map[string]any{
+			"LeafCertTTL":         "1h",
+			"PrivateKeyType":      tc.SigningKeyType,
+			"PrivateKeyBits":      tc.SigningKeyBits,
+			"RootPKIPath":         "pki-root/",
+			"IntermediatePKIPath": "pki-intermediate/",
+		})
 
-			{
-				root, err := provider2.GenerateRoot()
-				require.NoError(t, err)
-				assertCorrectKeyType(t, tc.CSRKeyType, root.PEM)
+		testutil.RunStep(t, "init", func(t *testing.T) {
+			root, err := provider1.GenerateRoot()
+			require.NoError(t, err)
+			assertCorrectKeyType(t, tc.SigningKeyType, root.PEM)
 
-				intPEM, err := provider2.ActiveIntermediate()
-				require.NoError(t, err)
-				assertCorrectKeyType(t, tc.CSRKeyType, intPEM)
+			intPEM, err := provider1.ActiveIntermediate()
+			require.NoError(t, err)
+			assertCorrectKeyType(t, tc.SigningKeyType, intPEM)
+		})
+
+		testVault2 := NewTestVaultServer(t)
+
+		attr2 := &VaultTokenAttributes{
+			RootPath:         "pki-root",
+			IntermediatePath: "pki-intermediate",
+			ConsulManaged:    true,
+			WithSudo:         false, // irrelevant for the new CA provider
+		}
+		token2 := CreateVaultTokenWithAttrs(t, testVault2.client, attr2)
+
+		provider2 := createVaultProvider(t, true, testVault2.Addr, token2, map[string]any{
+			"LeafCertTTL":         "1h",
+			"PrivateKeyType":      tc.CSRKeyType,
+			"PrivateKeyBits":      tc.CSRKeyBits,
+			"RootPKIPath":         "pki-root/",
+			"IntermediatePKIPath": "pki-intermediate/",
+		})
+
+		testutil.RunStep(t, "swap", func(t *testing.T) {
+			root, err := provider2.GenerateRoot()
+			require.NoError(t, err)
+			assertCorrectKeyType(t, tc.CSRKeyType, root.PEM)
+
+			intPEM, err := provider2.ActiveIntermediate()
+			require.NoError(t, err)
+			assertCorrectKeyType(t, tc.CSRKeyType, intPEM)
+
+			if expectFailure {
+				testCrossSignProvidersShouldFail(t, provider1, provider2)
+			} else {
+				testCrossSignProviders(t, provider1, provider2)
 			}
+		})
+	}
 
-			testCrossSignProviders(t, provider1, provider2)
+	for _, tc := range tests {
+		t.Run(tc.Desc, func(t *testing.T) {
+			t.Run("without sudo", func(t *testing.T) {
+				run(t, tc, false, true)
+			})
+			t.Run("with sudo", func(t *testing.T) {
+				run(t, tc, true, false)
+			})
 		})
 	}
 }
 
 func TestVaultProvider_SignIntermediate(t *testing.T) {
-
 	SkipIfVaultNotPresent(t)
 
+	t.Parallel()
+
 	tests := CASigningKeyTypeCases()
 
 	for _, tc := range tests {
-		tc := tc
```

Comment:
We actually need this now that we're running `t.Parallel()`, right?

Reply:
I did the `run(t, tc)` thing here instead.